spec.emu

```diff
@@ -15,37 +15,68 @@ contributors: Jordan Harband
 	<emu-alg>
 		1. Let _promise_ be the *this* value.
 		1. If IsPromise(_promise_) is *false*, throw a *TypeError* exception.
-		1. Let _thenFinally_ be ! CreateThenFinally(_onFinally_).
-		1. Let _catchFinally_ be ! CreateCatchFinally(_onFinally_).
+		1. Let _C_ be ? SpeciesConstructor(_promise_, %Promise%).
+		1. Let _thenFinally_ be ! CreateThenFinally(_C_, _onFinally_).
+		1. Let _catchFinally_ be ! CreateCatchFinally(_C_, _onFinally_).
 		1. Return ? Invoke(_promise_, `"then"`, &laquo; _thenFinally_, _catchFinally_ &raquo;).
 	</emu-alg>
 </emu-clause>
 
 <emu-clause id="sec-createthenfinally" aoid="CreateThenFinally">
-	<h1>CreateThenFinally ( _onFinally_ )</h1>
-	<p>The abstract operation CreateThenFinally takes an _onFinally_ function, and returns a callback function for use in Promise.prototype.finally.</p>
+	<h1>CreateThenFinally ( _C_, _onFinally_ )</h1>
+	<p>The abstract operation CreateThenFinally takes a Promise Constructor _C_, and an _onFinally_ function, and returns a callback function for use in Promise.prototype.finally.</p>
```

Comment:
You are inconsistent about casing: "Promise Constructor" here, "Promise constructor" below. However I don't think either are great phrasing, as they're too easily confused with "`Promise` constructor".

The existing spec says "a constructor function that supports the parameter conventions of the Promise constructor." That's pretty verbose. Maybe eventually we should refactor ES to define the term "`Promise`-like constructor" or something and refer to that.

In the meantime my current favorites are either lowercase "promise constructor" or (as of now undefined) "`Promise`-like constructor".